db/migrate/20220105085729_split_customers_name.rb

```diff
@@ -0,0 +1,39 @@
+class SplitCustomersName < ActiveRecord::Migration[6.1]
+  class Spree::DummyAddress < ApplicationRecord
+    self.table_name = 'spree_addresses'
+  end
+  class Customer < ApplicationRecord
+    belongs_to :bill_address, class_name: "Spree::DummyAddress"
+  end
+
+  def up
+    add_column :customers, :first_name, :string, null: false, default: ""
+    add_column :customers, :last_name, :string, null: false, default: ""
+    rename_column :customers, :name, :backup_name
+
+    migrate_customer_name_data!
+  end
+
+  def down
+    remove_column :customers, :first_name
+    remove_column :customers, :last_name
+    rename_column :customers, :backup_name, :name
+  end
+
+  def migrate_customer_name_data!
+    Customer.includes(:bill_address).where.not(bill_address_id: nil).find_each do |customer|
```

Comment:
If you do `joins(:bill_address)` then the inner join makes sure that you only get records with addresses.

Reply:
Sometimes I am worst than a machine...